src/ChunkMap.cpp

```diff
@@ -78,23 +64,19 @@ void cChunkMap::RemoveLayer(cChunkLayer * a_Layer)
 cChunkMap::cChunkLayer * cChunkMap::GetLayer(int a_LayerX, int a_LayerZ)
 {
 	cCSLock Lock(m_CSLayers);
-	for (cChunkLayerList::const_iterator itr = m_Layers.begin(); itr != m_Layers.end(); ++itr)
+	auto it = m_Layers.find(std::pair<int, int>(a_LayerX, a_LayerZ));
+	if (it !=m_Layers.end())
 	{
-		if (((*itr)->GetX() == a_LayerX) && ((*itr)->GetZ() == a_LayerZ))
-		{
-			return *itr;
-		}
+		cChunkLayer & Layer = it->second;
+		return &Layer;
 	}
 
 	// Not found, create new:
-	cChunkLayer * Layer = new cChunkLayer(a_LayerX, a_LayerZ, this, *m_Pool);
-	if (Layer == nullptr)
-	{
-		LOGERROR("cChunkMap: Cannot create new layer, server out of memory?");
-		return nullptr;
-	}
-	m_Layers.push_back(Layer);
-	return Layer;
+	m_Layers.emplace(std::piecewise_construct,
```

Comment:
Emplace return an iterator, I can get rid of the m_Layers.find  below, and gain some more performance.